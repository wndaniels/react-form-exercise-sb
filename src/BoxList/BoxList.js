import React, { useState } from "react";
import NewBoxForm from "../NewBoxForm/NewBoxForm";
import Box from "../Box/Box";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const addBox = (newBox) => {
    setBoxes((boxList) => [...boxList, { ...newBox }]);
  };

  const removeBox = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };

  return (
    <div>
      <h1>BoxList</h1>

      <NewBoxForm addBox={addBox} />
      <div>
        {boxes.map((box) => (
          <Box
            id={box.id}
            backgroundColor={box.backgroundColor}
            width={box.width}
            height={box.height}
            handleRemove={removeBox}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
