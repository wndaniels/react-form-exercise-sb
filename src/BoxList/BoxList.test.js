import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", () => {
  render(<BoxList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new box", () => {
  const { queryByText, getByLabelText } = render(<BoxList />);
  const color = getByLabelText("Color");
  const width = getByLabelText("Width");
  const height = getByLabelText("Height");
  const btn = queryByText("Add New Box");
  expect(queryByText("X")).not.toBeInTheDocument();
  fireEvent.change(color, { target: { value: "#000000" } });
  fireEvent.change(width, { target: { value: 16 } });
  fireEvent.change(height, { target: { value: 16 } });
  fireEvent.click(btn);
  expect(queryByText("X")).toBeInTheDocument();
});
