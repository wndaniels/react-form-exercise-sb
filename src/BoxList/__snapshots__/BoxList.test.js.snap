// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      BoxList
    </h1>
    <form>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="backgroundColor"
        type="color"
        value="#000000"
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        max="30"
        min="5"
        name="width"
        type="range"
        value="16"
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        max="30"
        min="5"
        name="height"
        type="range"
        value="16"
      />
      <button>
        Add New Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
